url.bs

```diff
@@ -2483,6 +2483,8 @@ interface URL {
            attribute USVString search;
   [SameObject] readonly attribute URLSearchParams searchParams;
            attribute USVString hash;
+
+  USVString toJSON();
```

Comment:
Is the question one of having `serializer;` basically be a thing that produces a then-custom-defined toJSON, kinda like `stringifier;`?

The problem is that unlike `stringifier;`, where the return value is clear, the return value of toJSON could be all sorts of things.  We could define `serializer;` to induce the equivalent of `any toJSON();` but that's more annoying to work with than being able to usefully declare your return type...